app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
+          return service.getters[getterId].kind === 'ThinkerbellRuleOn';
+        });
+        const sourceGetterId = Object.keys(service.getters).find(getterId => {
+          return service.getters[getterId].kind === 'ThinkerbellRuleSource';
+        });
+
+        const enabledSetterId = Object.keys(service.setters).find(setterId => {
+          return service.setters[setterId].kind === 'ThinkerbellRuleOn';
+        });
+        const removeSetterId = Object.keys(service.setters).find(setterId => {
+          return service.setters[setterId].kind === 'RemoveThinkerbellRule';
         });
 
-        return Promise.all(promises)
-          // Map all the recipes to a more user-friendly format.
-          .then(servicesEnabled => recipes.map((recipe, index) => ({
-            id: recipe.id,
-            label: recipe.id,
-            enabled: servicesEnabled[index][`${recipe.id}/get_enabled`]
-              && servicesEnabled[index][`${recipe.id}/get_enabled`]
-                .OnOff === 'On'
-          })));
+        return {
+          id: service.id,
+          getEnabled: enabledGetterId,
+          getSource: sourceGetterId,
+          setEnabled: enabledSetterId,
+          remove: removeSetterId,
+          status: null,
+          source: null
+        };
       });
-  }
+    });
 
-  getServicesWithGetters() {
-    return this[p.net].fetchJSON(
-      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
-      // Get all services but the recipes.
-      .then(services => services.filter(service =>
-        service.adapter !== 'thinkerbell-adapter'
-          && service.adapter !== 'webpush@link.mozilla.org'
-          && service.adapter !== 'console@link.mozilla.org'
-
-          // Deactivating IP camera from services with getters.
-          && service.adapter !== 'ip-camera@link.mozilla.org'
-      ))
-      // Keep only the services with getters.
-      .then(services => services.filter(service =>
-        Object.keys(service.getters).length
-      ))
-      // User friendly name.
-      .then(services => services.map(service => {
-          switch (service.adapter) {
-            case 'clock@link.mozilla.org':
-              service.name = 'Everyday';
-              break;
-
-            default:
-              service.name = service.adapter;
-              break;
+    // Fetch recipe enabled statuses and sources.
+    return recipeServicesPromise.then(services => {
+      // Fetch their respective enabled status.
+      const { enabledSelectors, sourceSelectors } = services.reduce(
+        (selectors, service) => {
+
+          selectors.enabledSelectors.push({ id: service.getEnabled });
+          selectors.sourceSelectors.push({ id: service.getSource });
+
+          return selectors;
+        },
+        { enabledSelectors: [], sourceSelectors: [] }
+      );
+
+      const getterURL = `${this[p.net].origin}/api/` +
+        `v${this[p.settings].apiVersion}/channels/get`;
+
+      return Promise.all([
```

Comment:
Getting all statuses and sources (to get name of the recipe) for all recipes in two requests :) Actually we can do it even in one, but it didn't work for me - need to discuss.